![](image.jpg)

Humans not only take debts to manage necessities. A country may also take debt to manage its economy. For example, infrastructure spending is one costly ingredient required for a country's citizens to lead comfortable lives. The World Bank is the organization that provides debt to countries.

In this project, you are going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. You are going to find the answers to the following questions:

- What is the number of distinct countries present in the database?
- What country has the highest amount of debt?
- What country has the lowest amount of repayments?

Below is a description of the table you will be working with:

## `international_debt` table

| Column | Definition | Data Type |
|-|-|-|
|country_name|Name of the country|`varchar`|
|country_code|Code representing the country|`varchar`|
|indicator_name|Description of the debt indicator|`varchar`|
|indicator_code|Code representing the debt indicator|`varchar`|
|debt|Value of the debt indicator for the given country (in current US dollars)|`float`|

You will execute SQL queries to answer three questions, as listed in the instructions.

# Analyze International Debt Statistics

## Question 1: Number of Distinct Countries

First, we want to find out how many unique countries are represented in the dataset.
This helps us understand the scope of the data coverage.

In [4]:
SELECT 
    COUNT(DISTINCT country_name) AS total_distinct_countries
FROM international_debt;


total_distinct_countries
0                       124

### Result:

Number of countries: 124

### Interpretation:

This indicates that the dataset covers 124 developing countries, providing broad coverage for debt analysis.

## Question 2: Country with the Highest Total Debt

Next, we calculate the total debt for each country to identify which country carries the largest debt burden.

In [5]:
SELECT 
    country_name, 
    SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY SUM(debt) DESC
LIMIT 1;


country_name    total_debt
0        China  2.857935e+11

### Result:

- Country: China
- Total Debt: approximately $285.793 billion

### Interpretation:
China has the highest total debt among developing countries, highlighting its significant debt burden compared to others.

## Question 3: Country with the Lowest Principal Repayments
Here, we focus on principal repayments using the indicator code DT.AMT.DLXF.CD.
This query identifies the country with the lowest principal repayment amount.

In [6]:
SELECT 
    country_name, 
    indicator_name,
    MIN(debt) AS lowest_repayment
FROM international_debt
WHERE indicator_code = 'DT.AMT.DLXF.CD'
GROUP BY country_name, indicator_name
ORDER BY lowest_repayment
LIMIT 1;


country_name  ... lowest_repayment
0  Timor-Leste  ...           825000

[1 rows x 3 columns]

### Result:
- Country: TimorLeste
- Indicator: Principal repayments on external debt, long-term
- Lowest Repayment: $825,000

### Interpretation:
Timor-Leste has the lowest principal repayment amount on long-term external debt, reflecting differences in debt structure and repayment obligations across countries.

## **Conclusion**

This analysis provides an overview of debt distribution among developing countries, identifying the most and least indebted nations in terms of total debt and principal repayments.
Using SQL aggregation and filtering techniques, I was able to extract clear insights from the data to support economic analysis.